test/test.pl

```diff
@@ -154,6 +210,37 @@ sub _cmd
     }
     return ($? >> 8, join('',@out));
 }
+sub _cmd3
+{
+    my ($cmd) = @_;
+
+    my $tmp = "$$opts{tmp}/tmp";
+    my $pid = fork();
+    if ( !$pid )
+    {
+        exec('bash', '-o','pipefail','-c', "($cmd) 2>$tmp.e >$tmp.o");
+    }
+    waitpid($pid,0);
+
+    my $status  = $? >> 8;
+    my $signal  = $? & 127;
+
+    my (@out,@err);
+    if ( open(my $fh,'<',"$tmp.o") )
+    {
+        @out = <$fh>;
```

Comment:
I'm not sure why we use an array for this (it's in other code too).  It would be more efficient to just read everything into a string:
```
local $/; # Read whole file
$out = <$fh>;
```

Reply:
This is just out of habit, usually I'd do something with the output per-line. I don't think this is a performance issue in these tests, but okay.